### ISSUE TITLE
Type resolution fails for class attributes with Callable[[], None] annotation

### ISSUE THREAD

**Issue by Gavin-Hofer** [microsoft/pyright#10728]
### Bug Summary

Pyright is unable to resolve class attributes annotated with Callable[[], None] (nullary callables) due to the following error: `Could not bind method "<anonymous>" because it is missing a "self" or "cls" parameter (reportAttributeAccessIssue)`.

### Expected Behavior

Class attributes with `Callable[[], None]` annotation should resolve to `() -> None` type signature, consistent with how static methods with no arguments are handled.

### Actual Behavior

Static type checking fails with the following error:

```
% pyright example.py
example.py
  example.py:18:11 - error: Cannot access attribute "my_nullary_func" for class "type[TestClass]"
    Could not bind method "<anonymous>" because it is missing a "self" or "cls" parameter (reportAttributeAccessIssue)
1 error, 0 warnings, 0 informations 
```

### Reproduction Case

```python
from typing import Callable


class TestClass:
    my_nullary_func: Callable[[], None] = lambda: None
    my_unary_func: Callable[[int], None] = lambda _x: None

    @staticmethod
    def my_nullary_staticmethod() -> None:
        return None

    @staticmethod
    def my_unary_staticmethod(_x: int) -> None:
        return None


# ❌ Resolves to Unknown
TestClass.my_nullary_func

# ✅ Resolves to (int) -> None
TestClass.my_unary_func

# ✅ Resolves to () -> None
TestClass.my_nullary_staticmethod

# ✅ Resolves to (int) -> None
TestClass.my_unary_staticmethod
```

### Environment

Python 3.10.14
macOS Sequoia Version 15.5

Python packages (requirements.txt):
```
nodeenv==1.9.1
nodejs-wheel-binaries==22.17.0
pathspec==0.12.1
pyright==1.1.403
typing_extensions==4.14.1
```

**Comment by erictraut** [comment]
Pyright is resolving the attribute, but it's not able to bind it to the class because it doesn't have a `self` parameter with a type that is compatible with the bound class. Binding is necessary here because, among other reasons, it is needed to specialize a generic callable. 

There's a recent discussion about this in the [Python typing forum](https://discuss.python.org/t/when-should-we-assume-callable-types-are-method-descriptors/92938). That thread has not (yet) converged on consensus, but it nicely articulates the challenges and explores some potential changes to the type system to improve this case. If you have opinions about a solution here, please feel free to respond to that thread.

I think it's unlikely that we'll change pyright's current behavior here unless/until the typing spec is updated to indicate some other behavior. Making a change to the behavior now without such guidance from the spec would be disruptive for pyright users because it would break other (more common) use cases. 

I'm going to keep the issue open for now and think about it more.
